tests/routes/test_datasets.py

```diff
@@ -26,82 +31,79 @@
 }
 
 
-def test_datasets(client, db):
+@pytest.mark.asyncio
+async def test_datasets(async_client):
     """Basic test to check if empty data api response as expected."""
 
     dataset = "test"
 
-    response = client.get("/meta")
+    response = await async_client.get("/datasets")
     assert response.status_code == 200
     assert response.json() == {"data": [], "status": "success"}
 
-    response = client.put(f"/meta/{dataset}", data=json.dumps(payload))
+    response = await async_client.put(f"/dataset/{dataset}", data=json.dumps(payload))
     assert response.status_code == 201
     assert response.json()["data"]["metadata"] == payload["metadata"]
 
-    response = client.get("/meta")
+    response = await async_client.get("/datasets")
     assert response.status_code == 200
     assert len(response.json()["data"]) == 1
     assert response.json()["data"][0]["metadata"] == payload["metadata"]
 
-    response = client.get(f"/meta/{dataset}")
+    response = await async_client.get(f"/dataset/{dataset}")
     assert response.status_code == 200
     assert response.json()["data"]["metadata"] == payload["metadata"]
 
-    cursor = db.execute(
-        "SELECT schema_name FROM information_schema.schemata WHERE schema_name = :dataset;",
-        {"dataset": dataset},
-    )
-    rows = cursor.fetchall()
+    async with ContextEngine("WRITE"):
+        rows = await db.all(
+            f"SELECT schema_name FROM information_schema.schemata WHERE schema_name = '{dataset}';"
+        )
+
     assert len(rows) == 1
 
     new_payload = {"metadata": {"title": "New Title"}}
-    response = client.patch(f"/meta/{dataset}", data=json.dumps(new_payload))
+    response = await async_client.patch(
+        f"/dataset/{dataset}", data=json.dumps(new_payload)
+    )
     assert response.status_code == 200
     assert response.json()["data"]["metadata"] != payload["metadata"]
     assert response.json()["data"]["metadata"]["title"] == "New Title"
     assert response.json()["data"]["metadata"]["subtitle"] == "string"
 
-    response = client.delete(f"/meta/{dataset}")
+    response = await async_client.delete(f"/dataset/{dataset}")
     assert response.status_code == 200
     assert response.json()["data"]["dataset"] == "test"
 
-    cursor = db.execute(
-        "SELECT schema_name FROM information_schema.schemata WHERE schema_name = :dataset;",
-        {"dataset": dataset},
-    )
-    rows = cursor.fetchall()
+    async with ContextEngine("WRITE"):
```

Comment:
Just to be sure I understand, this could be done with "READ" instead, right?

Reply:
It does. I wasn't sure if the read only permissions was brought enough to access those system tables (but never bothered to check)